### PR TITLE
[Backport][BUGFIX] - Timezone conversion fix for catalog module product option …

### PR DESCRIPTION
**Pull request by CsSatter**
…type date model getFormattedOptionValue method

### Description
Saved value in UTC is expected to be in local timezone on frontend.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### Original PR 
https://github.com/magento/magento2/pull/18180

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18181) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @CsSatter. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by aleron75 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @aleron75, thank you for the review.
[ENGCOM-3031](https://jira.corp.magento.com/browse/ENGCOM-3031) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@CsSatter thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by aleron75** [comment]
Hi @CsSatter, we need the fix to be delivered also to 2.3-develop branch so, please, don't forget to port this PR to 2.3-develop; you can do it easily using the Porting Tool, [read more here](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically)

**Comment by CsSatter** [comment]
Hi @aleron75 I created another [pull request](https://github.com/magento/magento2/pull/18180) for 2.3, but it didn't pass all checks. However after rerunning the build this morning it has passed all the checks.

**Comment by sidolov** [comment]
Hi @CsSatter , can you clarify the case when you need date converted to local timezone? 
I'm asking for the case because this changes especcially were made for cases that you described  (in  commit: 465f3ef4dedc6097b2a6aa22c8df959c2c383aa1). Customer shouldn't know about store timezone and date should be passed as is. 
Thank you!

**Comment by CsSatter** [comment]
Hi @sidolov , the case which required this change was a case of where a custom date option was configured to allow a customer to set a delivery date for his/her product. The input is automatically changed to a datetime type regardless of which configuration option is chosen. Because there is a time difference between the store timezone (Europe/Amsterdam) and UTC timezone the value gets converted to the previous date at 22:00:00. When the customer is viewing the cart we show the chosen delivery date again, however due to the timezone parameter being passed is always 'UTC' it doesn't get converted back to Europe/Amsterdam.

From my understanding it is common for Magento to convert its times to UTC to preserve them in case the timezone configuration is changed. Therefore we believe this change is required, changing the parameter from a hardcoded UTC to null allowed Magento to convert the datetime value to UTC but now also allowing it to be changed to its proper timezone for when it is required on the frontend.

In short:
1. Configuration for the store is Dutch locale and Europe/Amsterdam as the configured timezone
2. Product has a custom option configured for date
3. Customer declares a date (example date: 15-09-2018)
4. Magento converts and saves date to UTC (14-09-1995 22:00:00)
5. Customer goes to the cart to checkout
6. Magento converts saved date from UTC to configured timezone (expected, but instead converts saved date to UTC resulting in the value still being 14-09-1995 22:00:00)
7. Magento renders the value for the frontend
8. Rendered date is displayed as 14-09-1995 (15-09-1995 expected)

**Comment by sivaschenko** [comment]
Closing as 2.3 PR is closed due to inactivity
